process/coordinator/process.go

```diff
@@ -555,13 +555,18 @@ func (tc *transactionCoordinator) CreateMbsAndProcessCrossShardTransactionsDstMe
 	miniBlocks := make(block.MiniBlockSlice, 0)
 	nrTxAdded := uint32(0)
 	nrMiniBlocksProcessed := 0
+	processedTxHashes := make([][]byte, 0)
 
 	if check.IfNil(hdr) {
 		return miniBlocks, nrTxAdded, false, nil
 	}
 
 	shouldSkipShard := make(map[uint32]bool)
 
+	if tc.shardCoordinator.SelfId() == core.MetachainShardId {
+		tc.initProcessedTxsResults()
```

Comment:
why this ?

Reply:
Metachain should init post processor map which hold all scrs created after last init, right here, before starting to process miniblocks from the given shard hdr, just to keep track of all scrs created also from successful mbs processed. Then, if for example, the last mb processing will fail it could revert all the scrs coresponding to all mbs processed, no matter if they were successful or not.